kubespawner/spawner.py

```diff
@@ -2185,7 +2185,8 @@ async def poll(self):
         just Falsy, to determine that the pod is still running.
         """
 
-        await self._start_watching_pods()
+        reflector = await self._start_watching_pods()
+        await reflector.first_load_future
```

Comment:
This is the pattern I've seen elsewhere in the codebase, and I think that avoids some performance penalty of using `await`.

```suggestion
        if not reflector.first_load_future.done():
            await reflector.first_load_future
```

I'm not sure if it makes sense to do this here though, thinking maybe it belongs to `_start_watching_pods` or `_start_reflector`.

What do you think @ivyxjc?